va/va.go

```diff
@@ -401,7 +469,28 @@ func (va *ValidationAuthorityImpl) validateTLSSNI01(ctx context.Context, identif
 	Z := hex.EncodeToString(h.Sum(nil))
 	ZName := fmt.Sprintf("%s.%s.%s", Z[:32], Z[32:], core.TLSSNISuffix)
 
-	return va.validateTLSWithZName(ctx, identifier, challenge, ZName)
+	return va.validateTLSSNI01WithZName(ctx, identifier, challenge, ZName)
+}
+
+func (va *ValidationAuthorityImpl) validateTLSSNI02(ctx context.Context, identifier core.AcmeIdentifier, challenge core.Challenge) ([]core.ValidationRecord, *probs.ProblemDetails) {
+	if identifier.Type != "dns" {
+		va.log.Info(fmt.Sprintf("Identifier type for TLS-SNI-02 was not DNS: %s", identifier))
+		return nil, probs.Malformed("Identifier type for TLS-SNI-02 was not DNS")
+	}
+
+	// Compute the digest for the SAN b that will appear in the certificate
+	ha := sha256.New()
+	ha.Write([]byte(challenge.Token))
+	za := hex.EncodeToString(ha.Sum(nil))
```

Comment:
This and below can be simplified to `sha256.Sum256` instead of creating a new hash object each time.